diff --git a/extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmProcessor.java b/extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmProcessor.java
index 02a0485d..37fa83ba 100644
--- a/extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmProcessor.java
+++ b/extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmProcessor.java
@@ -832,7 +832,9 @@ public void build(RecorderContext recorderContext, HibernateOrmRecorder recorder
             integratorClasses.add((Class<? extends Integrator>) recorderContext.classProxy(integratorClassName));
         }
         if (launchMode.getLaunchMode() == LaunchMode.DEVELOPMENT) {
-            integratorClasses.add(HibernateOrmDevIntegrator.class);
+            // Use a class proxy so the class is resolved at runtime by the right classloader (e.g. in remote dev)
+            integratorClasses.add((Class<? extends Integrator>) recorderContext
+                    .classProxy(HibernateOrmDevIntegrator.class.getName()));
             integratorClasses.add(SchemaManagementIntegrator.class);
         }
 
