diff --git a/services/src/main/java/org/keycloak/protocol/oidc/endpoints/TokenEndpoint.java b/services/src/main/java/org/keycloak/protocol/oidc/endpoints/TokenEndpoint.java
index cf1e1584..40604498 100644
--- a/services/src/main/java/org/keycloak/protocol/oidc/endpoints/TokenEndpoint.java
+++ b/services/src/main/java/org/keycloak/protocol/oidc/endpoints/TokenEndpoint.java
@@ -36,6 +36,7 @@ import jakarta.ws.rs.core.MultivaluedMap;
 import jakarta.ws.rs.core.Response;
 import jakarta.ws.rs.core.Response.Status;
 
+import org.keycloak.OAuth2Constants;
 import org.keycloak.OAuthErrorException;
 import org.keycloak.common.ClientConnection;
 import org.keycloak.events.Details;
@@ -173,6 +174,12 @@ public class TokenEndpoint {
         try {
             return grant.process(context);
         } catch (TokenInterceptorException e) {
+            event.detail(REASON, e.getDescription());
+            String resource = formParams.getFirst(OAuth2Constants.RESOURCE);
+            if (resource != null) {
+                event.detail(OAuth2Constants.RESOURCE, resource);
+            }
+            event.error(e.getError());
             throw new CorsErrorResponseException(cors, e.getError(), e.getDescription(), Response.Status.BAD_REQUEST);
         } catch (RefreshTokenException e) {
             event.detail(REASON, e.getErrorDescription());
