diff --git a/integration/client-cli/admin-cli/src/main/java/org/keycloak/client/cli/util/IoUtil.java b/integration/client-cli/admin-cli/src/main/java/org/keycloak/client/cli/util/IoUtil.java
index ab277363..578de583 100644
--- a/integration/client-cli/admin-cli/src/main/java/org/keycloak/client/cli/util/IoUtil.java
+++ b/integration/client-cli/admin-cli/src/main/java/org/keycloak/client/cli/util/IoUtil.java
@@ -113,7 +113,9 @@ public class IoUtil {
         }
         if (!isRegularFile(path)) {
             createFile(path);
-            // make sure only owner can read/write it
+        }
+        {
+            // make sure only owner can read/write it, even if the file already existed
             if (FileSystems.getDefault().supportedFileAttributeViews().contains("posix")) {
                 setUnixPermissions(path);
             } else if (supportedViews.contains("acl")) {
